tools/scripts/lock-api-state.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash -ex
+
+# Buf image binary
+BUF_IMAGE=buf.image.bin
+PROJECT_ROOT=$(pwd)
+
+## lock in current protobuf state
+cd proto
+make gen-buf-image
+git add $BUF_IMAGE
```

Comment:
Actually, I think it would be better to just delete this line and use `-a` with `git commit`, then not even have a `BUF_IMAGE` variable at all, since that is a variable that has to be synchronized manually between multiple files.

Unless you really want to enforce that only the known BUF_IMAGE path is updated, but in that case you would have more work to do anyway, to assert that after `make gen-buf-image` there was only file updated (I doubt that's worth the effort).